File: download_data.js
const { Storage } = require('@google-cloud/storage');
const fs = require('fs');
const path = require('path');

// Create a storage client
const storage = new Storage();

async function downloadPublicFile(bucketName, srcFileName, destFileName) {
    const options = {
        destination: destFileName,
    };

    // Download public file.
    await storage.bucket(bucketName).file(srcFileName).download(options);

    console.log(
        `Downloaded public file ${srcFileName} from bucket name ${bucketName} to ${destFileName}`
    );
}

/**
 * Downloads all files from a public Google Cloud Storage bucket.
 * @param {string} bucketName - The name of the bucket.
 * @param {string} folder - The folder (prefix) in the bucket.
 * @param {string} destination - The local directory to save the files.
 */
async function downloadAllFilesFromBucket(bucketName, folder, destination) {
    try {
        // Ensure the destination directory exists
        if (!fs.existsSync(destination)) {
            fs.mkdirSync(destination, { recursive: true });
        }

        // Get the list of files in the bucket
        const [files] = await storage.bucket(bucketName).getFiles({ prefix: folder });

        // Download each file
        for (const file of files) {
            const destFileName = path.join(destination, file.name.replace(folder, ''));
            console.log(`Downloading ${file.name} to ${destFileName}...`);
            await file.download({ destination: destFileName });
            console.log(`Downloaded ${file.name}`);
        }

        console.log('All files downloaded successfully.');
    } catch (error) {
        console.error('Error downloading files:', error);
    }
}

downloadAllFilesFromBucket('quickdraw_dataset', 'full/numpy_bitmap', 'data/numpy_bitmap');
//downloadPublicFile('quickdraw_dataset', 'full/binary/angel.bin', './data/testangel.bin');

File: scripts/download_data.py
from google.cloud import storage
import os

def download_from_folder(bucket_name, folder_path, destination_folder):
    """
    Download files from a specific folder in a Google Cloud Storage bucket.

    :param bucket_name: Name of the GCS bucket.
    :param folder_path: The "folder" path in the bucket (e.g., "full/simplified/").
    :param destination_folder: Local directory to save the downloaded files.
    """
    # Initialize the anonymous client
    client = storage.Client.create_anonymous_client()
    bucket = client.bucket(bucket_name)
    blobs = bucket.list_blobs(prefix=folder_path)

    # Ensure the destination folder exists
    if not os.path.exists(destination_folder):
        os.makedirs(destination_folder)

    print(f"Downloading files from folder '{folder_path}' in bucket '{bucket_name}' to '{destination_folder}'")

    # Download files
    for blob in blobs:
        if blob.name.endswith("/"):  # Skip empty folder-like blobs
            continue

        # Define the local file path
        local_file_path = os.path.join(destination_folder, blob.name[len(folder_path):])

        # Create directories for nested paths
        os.makedirs(os.path.dirname(local_file_path), exist_ok=True)

        # Download the file
        blob.download_to_filename(local_file_path)
        print(f"Downloaded {blob.name} to {local_file_path}")

    print("Download complete.")


# Example usage
bucket_name = "quickdraw_dataset"
data_folders = ["binary/", "numpy_bitmap/", "raw/", "simplified/"]
folder_path = "full/" + data_folders[1]  # The folder you want to download
destination_folder = ".././data/numpy_bitmap"  # Destination folder
download_from_folder(bucket_name, folder_path, destination_folder)